addons/l10n_it_edi/models/account_edi_format.py

```diff
@@ -449,8 +477,40 @@ def _import_fattura_pa(self, tree, invoice):
         :param invoice: the invoice to update or an empty recordset.
         :returns:       the invoice where the fattura_pa data was imported.
         """
+        buyer_seller_info = self._l10n_it_buyer_seller_info()
+
         company = invoice.company_id
 
+        # There are 2 cases:
+        # - cron:
+        #     * Move direction (incoming / outgoing) flexible (no 'default_move_type')
+        #     * I.e. used for import from tax agency
+        # - "Upload" button (invoices / bills view)
+        #     * Fixed move direction; the button sets the 'default_move_type'
+        default_move_type = self.env.context.get('default_move_type')
+        if default_move_type is None:
+            incoming_possibilities = [True, False]
+        elif default_move_type in invoice.get_purchase_types(include_receipts=True):
+            incoming_possibilities = [True]
+        elif default_move_type in invoice.get_sale_types(include_receipts=True):
+            incoming_possibilities = [False]
+        else:
+            _logger.warning("Cannot handle default_move_type '%s'.", default_move_type)
+            return
+
+        for incoming in incoming_possibilities:
+            company_role, partner_role = ('buyer', 'seller') if incoming else ('seller', 'buyer')
+            company_info = buyer_seller_info[company_role]
+            elements = tree.xpath(company_info['vat_xpath'])
+            if elements and elements[0].text.casefold() in (company.vat or '').casefold():
```

Comment:
Just an info: in 17.0 you'll have the `get_text` function to extract text from elements, please use it in fw-port.

Reply:
Thanks I will